definitions/features/service.rb

```diff
@@ -50,6 +50,11 @@ def filter_disabled_services!(action, service_list)
     service_list
   end
 
+  def unit_file_available?(name)
+    exit_status, = execute_with_status("systemctl list-unit-files --type=service | grep -w #{name}")
```

Comment:
```suggestion
    exit_status, = execute_with_status("systemctl --no-legend --no-pager list-unit-files --type=service #{name} | grep -qw #{name}")
```

I think this is cleaner:
* `--no-legend` will suppress additional output from systemctl, which should not hurt, but you never know.
* `--no-pager` will disable piping through `less` and friends. `systemctl` *does* autodetect this, but I always prefer being explicit
* if you add the `#{name}` to the systemctl call, it will only list that one unit
* we don't care for the *output* of `grep`, so we can `-q` (quiet) it. and while we're at it, it's probably better to write `--word-regexp --quiet` instead.

And now after I said this, I wonder if it would make sense (not for the usecase at hand, but in future?) to actually store the whole list of unit files in f-maintain and match *all* services against that list?

Reply:
Do we even need `grep` here?
```console
$ systemctl --no-legend --no-pager list-unit-files --type=service doesnotexist
$ echo $?
1
$ systemctl --no-legend --no-pager list-unit-files --type=service systemd-resolved.service
systemd-resolved.service enabled enabled
$ echo $?
0
```

Reply:
@evgeni Thanks for description, changes are added accordingly. I like to know how keeping list of all unit files will help? As foreman-maintain cares about services which are defined under features.
@ekohl as we have discussed the systemctl does not return exit code 1 on failing to find the unit file on EL7 and Fedora 31. The Fedora 34 fixed this, however we are keeping the grep command for older version compatibility.

Reply:
Yep, given Fedora 31 doesn't have it, it's also safe to assume EL8 doesn't so it'll be a long time before we can count on it.